pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/MetadataBasedAggregationOperator.java

```diff
@@ -69,34 +69,26 @@ protected IntermediateResultsBlock getNextBlock() {
 
     for (AggregationFunctionContext aggregationFunctionContext : _aggregationFunctionContexts) {
       AggregationFunction function = aggregationFunctionContext.getAggregationFunction();
-      AggregationFunctionFactory.AggregationFunctionType functionType =
-          AggregationFunctionFactory.AggregationFunctionType.valueOf(function.getName().toUpperCase());
+      String functionName = function.getName();
       String column = aggregationFunctionContext.getAggregationColumns()[0];
-
       SegmentMetadataImpl segmentMetadata = (SegmentMetadataImpl) _segmentMetadata;
       AggregationResultHolder resultHolder;
-      switch (functionType) {
-        case COUNT:
-          resultHolder = new DoubleAggregationResultHolder(totalRawDocs);
-          break;
-        case MIN:
-          String minValue = segmentMetadata.getColumnMetadataFor(column).getMinValue().toString();
-          resultHolder = new DoubleAggregationResultHolder(Double.valueOf(minValue));
-          break;
-        case MAX:
-          String maxValue = segmentMetadata.getColumnMetadataFor(column).getMaxValue().toString();
-          resultHolder = new DoubleAggregationResultHolder(Double.valueOf(maxValue));
-          break;
-        case MINMAXRANGE:
-          String minValueRange = segmentMetadata.getColumnMetadataFor(column).getMinValue().toString();
-          String maxValueRange = segmentMetadata.getColumnMetadataFor(column).getMaxValue().toString();
-          resultHolder = new ObjectAggregationResultHolder();
-          resultHolder.setValue(new MinMaxRangePair(Double.valueOf(minValueRange), Double.valueOf(maxValueRange)));
-          break;
-
-        default:
-          throw new UnsupportedOperationException(
-              "Metadata based aggregation operator does not support function " + function.getName());
+      if (functionName.equals(AggregationFunctionType.COUNT.getName())) {
```

Comment:
you can still use the switch, just add a new method in the enum to return the functionType in of relying on valueOf.

Reply:
Basically that's the current **boolean matchFunctionName(String functionName)**. I'll change it to **getAggregationFunctionType** and return the type or **null**